openedx/core/djangoapps/programs/utils.py

```diff
@@ -424,3 +432,31 @@ def _attach_run_mode_upgrade_url(self, run_mode):
                 run_mode['upgrade_url'] = None
         else:
             run_mode['upgrade_url'] = None
+
+
+def course_start_datetime_text(start_date, advertised_start=None):
```

Comment:
I thought about that -- it seemed cleaner to just pass two args without trying to tease out the course_overview object. It's really a one-use function to patch these utils to continue to function as before, and it's only really a handler for the start_date. The end_date is a simple srftime.